### ISSUE TITLE
Error with code that contains the new experimental "examplesIf" tag

### ISSUE THREAD

**Issue by krlmlr** [r-lib/styler#742]
``` r
styler::style_text("
#' @examplesIf TRUE
#' NULL
NULL
")
#> Error: <text>:1:4: unexpected numeric constant
#> 1: If TRUE
#>        ^
```

<sup>Created on 2021-03-04 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

**Comment by krlmlr** [comment]
Also, it seems that if an .R file contains one single `#' @examplesIf`, it is not styled at all but a styling error is reported. Should we handle this more gracefully?

**Comment by lorenzwalthert** [comment]
Thanks. Re: fail gracefully: this is tracked in #732.

**Comment by lorenzwalthert** [comment]
Turns out this is more complicated than anticipated because 

* multiple `#' @examples` tags to end up in one `.Rd` file can't be processed together anymore since we must distinguish between `#' @examples` and `#' @examplesIf`.
* can't just always add a line break after `#' @examples` because condition should remain on same line.

I have some WIP on this. How likely is `#' @examplesIf` to make it beyond experimental? Because I don't want to put in 10h to see it go away in the next release...

**Comment by krlmlr** [comment]
Can you treat `#' @examplesIf xxx` like `#' @examples` for now to simplify the code?

The tag is proving useful: https://github.com/search?q=org%3Acran+examplesIf&type=code. I think it's to stay.

@gaborcsardi: Are there plans to remove/replace `@examplesIf` in roxygen2?

**Comment by gaborcsardi** [comment]
No, I think it will stay.

**Comment by lorenzwalthert** [comment]
I agree we can treat it very similar to `@examples` now #748 is merged to close #746 and #747.

**Comment by lorenzwalthert** [comment]
Are the following restrictions reasonable?

* there must be a line break between condition and code:
```r
@exampleIf {cond} 
{code}
```
* This implies that condition can be on separate line

```r
@exampleIf 
{cond} 
{code}
```
But all of it can't be on same line 
```r
@exampleIf {cond}  {code}
```
* if condition can span multiple lines
```r
@exampleIf {cond
is long} 
{code}
```

**Comment by krlmlr** [comment]
I think the condition must be on the same line?

**Comment by lorenzwalthert** [comment]
Makes sense. I described above is what `{styler}` *can* support currently (and the case you mention works, but it's more of a coincidence). If `{roxygen2}` supports a subset of it, it's fine. The other way round would be more problematic, i.e. if 
```
@examplesIf cond code
```
Would be allowed with `{roxygen2}`, I had to adapt.

**Comment by lorenzwalthert** [comment]
Closed with #743.
